### ISSUE TITLE
Salination plant getting lithium in the input tank

### ISSUE THREAD

**Issue by xenoflot** [aidancbrady/Mekanism#2694]
Hi Aidan,

I set up some salination plants last night and found at breakfast time and again after work that there was a couple of mB of lithium in the input tank.

I'm using EnderIO ender fluid conduits. Conduits were configured (I'm pretty sure) to Insert only. I have a conduit drawing out lithium into a drum. The setup was in Woodland (BoP is installed, not sure if that's relevant) biome with normal daynight cycle.

I rebuilt them in a void age with all-day noon (although Rain still active) so the production seems non-stop and the issue has not raised its head again for a few hours.

I'm sure all the conduits were set to Insert only but I suppose there's a small chance that one of them was on In/Out which could have caused the pollution IF lithium can be re-inserted into the plant.

Cheers,
Xeno

**Comment by Cracks777** [comment]
it happens to me when my pumps dont keep up, the water runs out and puts the lithium in instead

**Comment by aidancbrady** [comment]
Right - you could solve this by making sure your pipes are set into push/pull mode based on what you intend for them to do.  You can use a Gauge Dropper to extract the few mb of whatever fluid mistakenly got in the plant.

**Comment by xenoflot** [comment]
I set it back up again after posting this and ensured that all pipes were one-way. Brine pipes in and lithium pipe out. I also tested that I was unable to pipe lithium back in (I wasn't) yet it still happened. As Cracks777 mentioned, this appears to only happen if the water supply allows the input tank to drain. Therefore it's not the pipes that are resulting in lithium getting into the input tank, but rather something internal to the salination plant.

Is this intended behaviour?
